openforcefield/utils/toolkits.py

```diff
@@ -693,6 +694,11 @@ def from_openeye(oemol, allow_undefined_stereo=False):
         from openeye import oechem
         from openforcefield.topology.molecule import Molecule
 
+
+        # Add explicit hydrogens if they're implicit
+        if oechem.OEHasImplicitHydrogens(oemol):
+            oechem.OEAddExplicitHydrogens(oemol)
```

Comment:
I'm not entirely sure how we should do this, actually. Does this change make `from_smiles(..., hydrogens_are_explicit=True)` (which is the default) behave unexpectedly? I think that function calls `from_openeye()`. Maybe `from_openeye()` should take an argument to control this behavior as well.

Reply:
Wow. Great thinking, I had totally missed the possible consequences there.

So, the case where this would be a problem is if the user provided a SMILES with a specific explicit protonation state, and the toolkit misinterpreted it as having implicit hydrogens (even though the user set `hydrogens_are_explicit=True`).

To prevent this from happening, I've updated both `OpenEyeToolkitWrapper` and `RDKitToolkitWrapper`'s `from_smiles` functions to assert that, if `hydrogens_are_explicit=True`, then there are zero implicit hydrogens on the molecule.